### PR TITLE
Converting TX API to use Hex strings

### PR DESCRIPTION
**Pull request by vimukthi-git**
<This is intended as a guide when describing-reviewing pull requests for go-centrifuge>
Issue Link: #692

### CODE REVIEW

File: transactions/transaction.go [hunk 1]
```diff
@@ -36,9 +51,53 @@ func NewLog(action, message string) Log {
 	}
 }
 
+// TxID is a centrifuge transaction ID. Internally represented by a UUID. Externally visible as a byte slice or a hex encoded string.
+type TxID uuid.UUID
```
Comment (vimukthi-git) [inline transactions/transaction.go]: main change

**Comment by vimukthi-git** [comment]
@vedhavyas its needed to avoid cyclic dependancies, if you know of a way for it to work without, I would welcome that.

**Comment by vimukthi-git** [comment]
@vedhavyas Thanks for the discussion, What I would take from it is that,
- We need to refactor the `configstore.RetriveConfig`(A bad design by me :P ) call so that it doesn't create dependancies everywhere. (Follow up pr)
- Since transaction package is needed everywhere(including `contextutil`) and transaction package may end up needing accounts and identity lets keep the new separate package structure inside transactions package. I find that the model(interface + sub package imple) that I found in my learnings to work seems to have been found by other people such as here https://go-talks.appspot.com/github.com/benbjohnson/structuring-applications-for-growth/main.slide#6 https://codeandtalk.com/v/gophercon-denver-2016/gophercon-2016-lightning-talk-ben-johnson

File: documents/purchaseorder/model_test.go [hunk 2]
```diff
@@ -8,38 +8,32 @@ import (
...
-	"github.com/stretchr/testify/mock"
-
-	"github.com/centrifuge/go-centrifuge/storage/leveldb"
-
-	"github.com/centrifuge/go-centrifuge/testingutils/config"
-
-	"github.com/centrifuge/go-centrifuge/identity/ethid"
-
-	"github.com/centrifuge/go-centrifuge/config/configstore"
-
```
Comment (vedhavyas) [inline documents/purchaseorder/model_test.go]: Thank you for arranging the imports

Comment (vimukthi-git) [reply documents/purchaseorder/model_test.go]: you made me have OCD about these now :D

**Pull request merged**